bridge/tracker/tracker.go

```diff
@@ -0,0 +1,390 @@
+package tracker
+
+import (
+	"context"
+	"encoding/json"
+	"math/big"
+	"os"
+
+	"github.com/0xPolygon/polygon-edge/types"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/syndtr/goleveldb/leveldb"
+	"github.com/umbracle/go-web3"
+)
+
+const (
+	lastQueriedBlockNumber = "last-block-num" //	db key
+)
+
+//	Tracker represents an event listener that notifies
+//	for each event emitted on the rootchain (ethereum).
+//	Events of interest are defined in rootchain.go
+type Tracker struct {
+	logger hclog.Logger
+
+	// required block confirmations
+	confirmations uint64
+
+	// newHeads subscription channel
+	headerCh chan *ethHeader
+
+	// event channel
+	eventCh chan []byte
+
+	// cancel funcs
+	cancelSubscription, cancelEventTracking context.CancelFunc
+
+	//	rootchain client
+	client *rootchainClient
+
+	//	db for last processed block number
+	db *leveldb.DB
```

Comment:
Why do we use LevelDB here specifically? Why not use the DB ambiguous object, that way when we add support for additional DB engines, we can just as easily swap them out

What do you think @lazartravica @Kourin1996 ?

Reply:
@Kourin1996 is right - we need a db to store the last processed block. Imagine the node goes down and you bring it back up again: we wouldn't want to query the same blocks again.

@zivkovicmilos you mean like a `store` interface ?

Reply:
@dbrajovic I don't think we should initialize database in this struct because it's hard to switch storage when testing or changing database.

Reply:
Of course we need storage, I'm just saying to use the DB store interface instead of a specific DB engine like LevelDB. Under the hood the implementation can be whatever we want. You're reinventing the wheel in this PR by writing additional code to manage LevelDB, when we already have support for it in the storage package.

Reply:
[Refactored](https://github.com/0xPolygon/polygon-edge/pull/386/commits/cea3677cb23fb747d347d6ea99d0ea2a9a490dcb)